apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -544,42 +651,200 @@ maps_contract(Config) ->
+
+    %% Map.size
+    %% Both of these still contain 3 elements.
+
+    {ISValue,_} = call_compute_func(NodeName, BPubkey, BPrivkey,
+                                    EncodedMapsPubkey,
+                                    <<"size_state_i">>, <<"()">>),
+    #{<<"value">> := 3} = decode_data(<<"int">>, ISValue),
+    {SSValue,_} = call_compute_func(NodeName, BPubkey, BPrivkey,
+                                    EncodedMapsPubkey,
+                                    <<"size_state_s">>, <<"()">>),
+    #{<<"value">> := 3} = decode_data(<<"int">>, SSValue),
+
+    %% Map.to_list, Map.from_list then test if element is there.
+
+    call_compute_func(NodeName, DPubkey, DPrivkey, EncodedTestPubkey,
+                      <<"list_state_i">>, <<"(242424)">>),
+    1 = call_member_state_i(NodeName, DPubkey, DPrivkey, EncodedMapsPubkey,
+                            <<"(242424)">>),
+    call_compute_func(NodeName, DPubkey, DPrivkey, EncodedTestPubkey,
+                      <<"list_state_s">>, <<"(\"xxx\")">>),
+    1 = call_member_state_s(NodeName, DPubkey, DPrivkey, EncodedMapsPubkey,
+                            <<"(\"xxx\")">>),
 
     ok.
 
-call_get_state(NodeName, Pubkey, Privkey, EncodedContractPubkey, HexCode) ->
+call_member_state_i(NodeName, Pubkey, Privkey, EncodedMapsPubkey, MemberArg) ->
+    call_func_decode(NodeName, Pubkey, Privkey, EncodedMapsPubkey,
+                     <<"member_state_i">>, MemberArg, <<"bool">>).
+
+call_member_state_s(NodeName, Pubkey, Privkey, EncodedMapsPubkey, MemberArg) ->
+    call_func_decode(NodeName, Pubkey, Privkey, EncodedMapsPubkey,
+                     <<"member_state_s">>, MemberArg, <<"bool">>).
+
+call_get_state(NodeName, Pubkey, Privkey, EncodedMapsPubkey) ->
     StateType = <<"( map(int, (int, int)), map(string, (int, int)) )">>,
-    {GSValue,_} = call_func(NodeName, Pubkey, Privkey, EncodedContractPubkey,
-			    HexCode, <<"get_state">>, <<"()">>),
-    #{<<"value">> := GetState} = decode_data(StateType, GSValue),
+    {Return,_} = call_compute_func(NodeName, Pubkey, Privkey,
+                                   EncodedMapsPubkey,
+                                   <<"get_state">>, <<"()">>),
+    #{<<"value">> := GetState} = decode_data(StateType, Return),
     GetState.
 
+call_get_state_i(NodeName, Pubkey, Privkey, EncodedMapsPubkey, GetArg) ->
+    {GSValue,_} = call_compute_func(NodeName, Pubkey, Privkey,
+                                    EncodedMapsPubkey,
+                                    <<"get_state_i">>, GetArg),
+    #{<<"value">> := GetValue} = decode_data(<<"(int, int)">>, GSValue),
+    GetValue.
+
+call_get_state_s(NodeName, Pubkey, Privkey, EncodedMapsPubkey, GetArg) ->
+    {GSValue,_} = call_compute_func(NodeName, Pubkey, Privkey,
+                                    EncodedMapsPubkey,
+                                    <<"get_state_s">>, GetArg),
+    #{<<"value">> := GetValue} = decode_data(<<"(int, int)">>, GSValue),
```

Comment:
Does this term returned by `decode_data` have a `<<"type">>` field? If yes, could you please assert what the value is? In the source code it is something like `{x = 5, y = 6}`.

I see that the [memory layout of Sophia encodes a record as the tuple of values](https://github.com/aeternity/protocol/blame/f8b99f42ea894d059a58ec4ebee2ce976415b6d8/contracts/sophia.md#L880) hence I expect the info on it being a record and the field names are lost. I understand this is why you specified `(int, int)`.